src/MLabsPAB/PreBalance.hs

```diff
@@ -158,7 +159,9 @@ balanceNonAdaOuts ownPkh utxos tx =
             txOuts
           (txOut@TxOut {txOutValue = v} : txOuts, txOuts') ->
             txOut {txOutValue = v <> nonAdaChange} : (txOuts <> txOuts')
-   in Right $ tx {txOutputs = outputs}
+   in if Value.isZero nonAdaChange
```

Comment:
It doesn't look like this function can fail, does it need to be in an Either context?

Reply:
Well, I just didn't want to break the pattern here:
https://github.com/mlabs-haskell/mlabs-pab/blob/e99af5435701a87f1df8791cd96e94d0e90355f7/src/MLabsPAB/PreBalance.hs#L50-L54
But yeah, you're right, I'm going to remove this